lib/engine/action/price_protect.rb

```diff
@@ -0,0 +1,10 @@
+# frozen_string_literal: true
+
+require_relative 'base'
+
+module Engine
```

Comment:
why does this need a new action? can't it just be a buy share?

Reply:
Indeed it doesn't. It didn't occur to me I could use the "same action" on a different step